feat: update security page to be company specific (#50)
shorebirdtech/handbook: src/content/docs/Company/security.md

### Employee Offboarding

When an employee leaves the company, their access to all systems is revoked
immediately. We have a checklist for offboarding employees that includes
revoking access to all systems. Since all access to all production systems is
gated on Google SSO, this is a simple process.

Access to our systems is reviewed regularly. So far our company is small enough
and we use SSO for all access, so this is a trivial process.

## Contractor Security & Confidentiality

Shorebird does not use contractors at this time. If we do in the future, they
will be covered by the same security policies and processes as employees and
sign the same confidentiality agreement.

## Internal Access Control

### Network

Shorebird does not run it's own internal network. Given that we are an all
[remote company](/company/company_details/#a-note-on-remote-only) this is of no
benefit to us and instead rely on Cloud Services.

We use HTTPS for all communication between our employees, our products, and our
customers. We use Google Cloud's managed SSL certificates for this.

### Devices

All employees are provided with company devices for work. If you believe you

> need to use a non company device for job related functions please work with the

Reviewer: ```suggestion
need to use a external device for job related functions please work with the
```

emails, or in any other insecure location. Passwords should never be passed to a
command line program as an argument, as they will then be stored in the shell
history file.

We do not recommend ever typing a password. Instead, we recommend using a
password manager to generate and store passwords. This ensures that passwords
will never be input into a place they were not intended to be.

Access to our SSO provider (Google) has required strong passwords (Google's
guidelines) and two factor authentication (2FA).

### Physical Access

Shorebird is an
[all-remote company](/company/company_details/#a-note-on-remote-only). We have
no physical office or visitor policies. We do not own or operate physical
servers.

### User Access Review

We review all user access to our systems periodically, as well as as part of an
employee joining or leaving the company. All access to Shorebird systems is
gated through Google SSO including required two factor authentication.

## Third Party Suppliers

We use a number of third party services to run our business. We list those which
may come in contact with customer data as part of our privacy policy:
https://shorebird.dev/privacy

> ### Third Party Service Accounts

Reviewer 2: Confirming this is not duplicated in the other doc?

Author: Good catch. I think having it in the handbook makes more sense as it's an overarching statement about services we use.

Author: Removed from docs via https://github.com/shorebirdtech/docs/pull/357

Accounts are bucketed into having handling customer data (e.g. Google Cloud) or
not (e.g. Canva).

- All accounts and services we use are logged in our Accounts sheet.
- All accounts should authenticate through Google OAuth (SSO). Exceptions must
  be approved by the CEO.
- Accounts that can never reach customer data (e.g. Twitter), may alternatively
  use a strong password stored in 1Password and two factor authentication if SSO
  is not available.
- Additions to the Accounts sheet are reviewed by the CEO.

## Incident Response

> We have a private playbook for incident response. We have logging and alerting

Reviewer: Might want to link to a status page 🤷 
We currently have https://shorebird.statuspage.io/ but haven't really been using it much and can easily switch to a different one.

in place to detect and respond to incidents. We have both dedicated private
channels on Discord for response as well as back-up text communication pathways
as well as phone numbers for all engineers.

We do not currently have separate incident tracking beyond our public GitHub. We
always notified all customers when affected by incidents (security or otherwise)
via their billing email address in the past and will continue to do so going
forward.

> ## Vendor Certifications

Reviewer 2: We should just change this to say ISO 27001 is in progress, expected completion mid 2025.

Shorebird maintains no vendor certifications at this time. We do from time to
time have security teams reach out and provide feedback on our APIs or source
code (which mostly public on [GitHub](https://github.com/shorebirdtech)).
Feedback always welcome.

> We will begin our ISO 27001 certifications in the near future.

Author: ```suggestion
We have started our ISO 27001 certification with an expected completion of Q3 2025.
```

## Product Production System Access

We use [Google Cloud IAM](https://cloud.google.com/iam) for access control and
[Google Cloud Logging](https://cloud.google.com/logging) for logging.

> A small number of engineers have access to production systems, for which we have
> a dedicated machine for access. Production changes are all done via CI/CD
> pipelines, as detailed in the Change Management section.

Reviewer: ```suggestion
A small number of engineers have access to production systems. Production changes are all done via CI/CD
pipelines, as detailed in the Change Management section.
```

> We have an additional (read-only) admin layer to a subset of our production

Reviewer 2: This is no longer read-only, but it has very limited support-only actions it can take.

> We have an additional (read-only) admin layer to a subset of our production

Author: ```suggestion
We have an additional admin layer to a subset of our production
```

systems for monitoring and support purposes.

## Business Continuity Planning

Shorebird has no formal Business Continuity Plan at this time.

Our code push product is designed such that any interruption to Shorebird's
services will not affect the users of your application, other than that you are
no longer able to provide them patches through Shorebird during such an
interruption. Shorebird is designed so that using Shorebird should never be
worse than not using Shorebird.

Not only is this good hygiene for our system, but it is also necessary since we
provide service to mobile applications which have unreliable network
connectivity and must therefore function well regardless of Shorebird
availability.

We monitor Shorebird's availability and have seen no interruption in Shorebird's
services in over a year. This is in large part due to our reliance on public
cloud infrastructure (Google, Cloud Flare) which themselves maintain high
degrees of reliability and business continuity planning.

> https://shorebird.statuspage.io/

Reviewer: Oh nice you linked it below 💯
